### DISCUSSION TITLE
Can't override button configuration when using `renderTag`

### DISCUSSION THREAD

**Discussion by webrgp** [solspace/craft-freeform#1553] [Questions & Answers] [answered]
### What happened?

When rendering a form using `renderTag` and `renderClosingTag`, the `buttons` configuration is not being passed to the form object.

The following works as expected:

```twig
{% set form = freeform.form('demo') %}

{{ form.render({
    buttons: {
        attributes: {
            submit: {
                class: 'foo',
            },
        },
    },
}) }}
```

But the following won't work:

```twig
{% set form = freeform.form('demo') %}

{{ form.renderTag({
    buttons: {
        attributes: {
            submit: {
                class: 'foo',
            },
        },
    },
}) }}
    ...
{{ form.renderClosingTag }}
```

I've got around this issue in the past with this work around:

```twig
{% set form = freeform.form('demo') %}

{{ form.renderTag({
    disable: ["submitButtons"]
}) }}
    ...
    {% set page = form.pages.current %}

    <div>
        <div>
            {# Custom placement of submit button #}
            {{ page.buttons.renderSubmit({
                class: 'foo',
            }) }}
        </div>
    </div>
{{ form.renderClosingTag }}
```

But now prevents me from correctly render anything that renders using the `EVENT_RENDER_BEFORE_CLOSING_TAG` event. For now I am using [Retcon](https://github.com/mmikkel/Retcon-Craft) to add my classes to the submit button, but a more permanent solution is needed.

When I inspect the `$form` variable in the [renderButtons](https://github.com/solspace/craft-freeform/blob/510d3b6e674d845627aa1f33cebbc4cc7680fdb2/packages/plugin/src/Bundles/Form/SubmitButtons/SubmitButtons.php#L98), I can see the configuration is being passed to the form's `$propertyBag`.

But when I inspect the `$page` [variable](https://github.com/solspace/craft-freeform/blob/510d3b6e674d845627aa1f33cebbc4cc7680fdb2/packages/plugin/src/Bundles/Form/SubmitButtons/SubmitButtons.php#L103), the form I get from the `$page->getForm()` method does not appear to have the configuration in the `$propertyBag`.

Overriding other props like `attributes`, `fields` and `captchas` work as expected.

### How can we reproduce this?

1. Create a form with a submit button
2. Render the form using `renderTag` and `renderClosingTag`
3. Pass a configuration to the `buttons` prop

### Freeform Edition

Pro

### Freeform Version

5.5.8

### Craft Version

4.11.5

### When did this issue start?

After upgrading from older Freeform version

### Previous Freeform Version

4.1.14

**Comment by kjmartens** [comment, marked as answer]
I'm sorry for the delay @webrgp,

I tried testing this on my end in Freeform 5.6.3 and it seems to always pass the class to the Submit button for me. Is this still happening for you? 🙂

**Reply by webrgp** [reply]
Yes. I will try updating to 5.6.3 to see if solves it.
